modules/spring-config/src/main/java/org/mule/runtime/config/spring/ModuleDelegatingEntityResolver.java

```diff
@@ -35,14 +37,19 @@
 
   private final Optional<ExtensionManager> extensionManager;
   private final EntityResolver entityResolver;
-  private final SchemaResourceFactory schemaResourceFactory;
+  private Optional<SchemaResourceFactory> schemaResourceFactory;
 
   public ModuleDelegatingEntityResolver(Optional<ExtensionManager> extensionManager) {
     this.entityResolver = new DelegatingEntityResolver(Thread.currentThread().getContextClassLoader());
     this.extensionManager = extensionManager;
     ServiceRegistry spiServiceRegistry = new SpiServiceRegistry();
-    schemaResourceFactory =
-        spiServiceRegistry.lookupProvider(SchemaResourceFactory.class, getClass().getClassLoader());
+    try {
+      // TODO(fernandezlautaro): MULE-11024 until the implementation is moved up to extensions-api, we need to work with Optional to avoid breaking the mule testing framework (cannot add the dependency, as it will imply a circular dependency)
+      schemaResourceFactory =
+          of(spiServiceRegistry.lookupProvider(SchemaResourceFactory.class, getClass().getClassLoader()));
```

Comment:
LookupProvider will throw IllegalStateException when there is zero or multiple providers. Setting empty is not the proper way to manage this situation. I think that you should use getProviders instead and manage the different results